nengo_loihi/conv.py

```diff
@@ -156,7 +156,11 @@ def build_conv2d_connection(model, conn):
     assert conn.post_slice == slice(None)
 
     gain = model.params[conn.post_obj.ensemble].gain
-    assert np.all(gain == gain[0]), "All gains must be the same"
+    if not np.all(gain == gain[0]):
+        # TODO: support this
+        raise NotImplementedError(
```

Comment:
I'm not sure how we could ever support this. If each neuron has a different gain, then there's no way to fold those gains into the convolution weights (since the essence of convolution is that neurons at different locations have the same weights).

Reply:
I'll change it to a validation error instead, to indicate that it probably won't change anytime in the near future.

Reply:
Done!